### PR TITLE
feat: unified event tasker

### PR DESCRIPTION
**Pull request by inureyes**
* This PR implements core components of unified tasker (#498)

## Implemented
 * Global event tasker
    * Task class format
    * Task pool with handlers
    * Garbage collector
 * Global progress indicator pool
 * Migrate old codes to user global objects

## Not implemented
  * Task monitor UI component (will be handled as separate issue)

### CODE REVIEW

File: src/components/backend-ai-indicator.ts [hunk 1]
```diff
@@ -45,19 +46,25 @@ export default class BackendAIIndicator extends LitElement {
...
-    this.text = 'Initializing...';
     this.mode = mode;
+    await this.updateComplete;
     this.dialog.show();
   }
 
   set(value, text = '') {
     this.value = value / 100.0;
     this.text = text;
+    if (value >= 1) {
```
Comment (adrysn) [inline src/components/backend-ai-indicator.ts]: Shouldn't it be `this.value`, not `value`?

File: src/components/backend-ai-tasker.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+ */
+
+import {css, customElement, html, LitElement, property} from "lit-element";
+
+class Task {
+  taskid: string;
+  taskobj: Object;
+  status: string;
+  created_at: number;
+
```
Comment (adrysn) [inline src/components/backend-ai-tasker.ts:14]: What about adding `finished_at` property as well, for possible future use before garbage collection.

File: src/components/backend-ai-indicator.ts (hunk 1 above)
Comment (inureyes) [reply src/components/backend-ai-indicator.ts]: Fixed at 413c09ce498

File: src/components/backend-ai-tasker.ts (hunk 2 above)
Comment (inureyes) [reply src/components/backend-ai-tasker.ts:14]: Fixed at 6aa5765

**Pull request merged**